docs/trace/extending-the-sdk/README.md

```diff
@@ -211,6 +212,14 @@ Writing an instrumentation library typically involves 3 steps.
          > Changing the name of the source should be considered a
          breaking change.
 
+    > **Warning**
+    > If enabling the instrumentation requires references to objects
+    from the targeted library, your instrumentation library should
+    offer public APIs to receive them instead of relying on the
+    extension methods, as objects might not be available yet at
```

Comment:
```suggestion
    offer public APIs to receive them instead of relying on the extension methods
    on `TracerProviderBuilder` and/or `IServiceCollection`, as the objects might not be available yet at
```